lib/record_store/provider.rb

```diff
@@ -19,6 +19,8 @@ def provider_for(zone_name)
           'DynECT'
         when /googledomains\.com\z/
           'GoogleCloudDNS'
+        when /.nsone\.net\z/
```

Comment:
if using the `.`, please backslash-it !
```suggestion
        when /\.nsone\.net\z/
```